arch/risc-v/src/mpfs/Kconfig

```diff
@@ -58,6 +59,11 @@ config MPFS_OPENSBI
         ---help---
                 This uses a ld-envm-opensbi.script linker script and the mpfs_opensbi.c code to use external OpenSBI.
 
+config BOARD_OPENSBI_DOMAINS
```

Comment:
Make this just "OPENSBI_DOMAINS" (CONFIG_OPENSBI_DOMAINS) and move the selection into arch/risc-v/src/opensbi/Kconfig

This is not just MPFS specific, but could be used with any RISC-V target, in theory